src/serializers/fields.rs

```diff
@@ -238,28 +283,75 @@ impl GeneralFieldsSerializer {
-            let (key, value) = result.map_err(py_err_se_err)?;
-            if extra.exclude_none && value.is_none() {
-                continue;
+        if extra.sort_keys {
+            let mut items = main_iter
+                .map(|r| -> PyResult<_> {
+                    let (k, v) = r?;
+                    let k_str = key_str(&k)?.to_owned();
+                    Ok((k_str, k, v))
+                })
+                .collect::<PyResult<Vec<_>>>()
+                .map_err(py_err_se_err)?;
+            items.sort_by(|(a, _, _), (b, _, _)| a.cmp(b));
+            for (key_str, key, value) in items {
+                self.process_serde_field::<S>(&key_str, &key, &value, &mut map, include, exclude, &extra)?;
             }
-            let key_str = key_str(&key).map_err(py_err_se_err)?;
-            let field_extra = Extra {
-                field_name: Some(key_str),
-                ..extra
-            };
+        } else {
+            for result in main_iter {
+                let (key, value) = result.map_err(py_err_se_err)?;
+                if extra.exclude_none && value.is_none() {
+                    continue;
+                }
+                let key_str = key_str(&key).map_err(py_err_se_err)?;
+                self.process_serde_field::<S>(key_str, &key, &value, &mut map, include, exclude, &extra)?;
+            }
+        }
+        Ok(map)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn process_serde_field<'py, S: serde::ser::Serializer>(
+        &self,
+        key_str: &str,
+        key: &Bound<'py, PyAny>,
+        value: &Bound<'py, PyAny>,
+        map: &mut S::SerializeMap,
+        include: Option<&Bound<'py, PyAny>>,
+        exclude: Option<&Bound<'py, PyAny>>,
+        extra: &Extra,
+    ) -> Result<(), S::Error> {
+        if extra.exclude_none && value.is_none() {
+            return Ok(());
+        }
 
-            let filter = self.filter.key_filter(&key, include, exclude).map_err(py_err_se_err)?;
-            if let Some((next_include, next_exclude)) = filter {
-                if let Some(field) = self.fields.get(key_str) {
-                    if let Some(ref serializer) = field.serializer {
-                        if !exclude_default(&value, &field_extra, serializer).map_err(py_err_se_err)? {
+        let field_extra = Extra {
+            field_name: Some(key_str),
+            ..*extra
+        };
+
+        let filter = self.filter.key_filter(key, include, exclude).map_err(py_err_se_err)?;
+        if let Some((next_include, next_exclude)) = filter {
+            if let Some(field) = self.fields.get(key_str) {
+                if let Some(ref serializer) = field.serializer {
+                    if !exclude_default(value, &field_extra, serializer).map_err(py_err_se_err)? {
+                        // Get potentially sorted value
+                        if extra.sort_keys {
+                            let sorted_dict = sort_dict_recursive(value.py(), value).map_err(py_err_se_err)?;
```

Comment:
Why do we still need to `sort_dict_recursive` here if we're already doing that in the dict serialization itself? (Same question for the 2 calls below)

Reply:
I try ur recommmendation and it doesn't work.

It will only work when converting to Python type (`to_python`). Since we wrote the serializing logic.
As for `to_json` we use the serde package if am not wrong.
They are separate into defined field processing and non defined fields processing. Cause we need to sort before using the serde package. 

This is based on my high level understanding.